paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java

```diff
@@ -433,6 +435,27 @@ public ProcessSystemDatabaseException() {
         }
     }
 
+    /** Exception for trying to operate on a table that doesn't have permission. */
+    class TableNoPermissionException extends Exception {
```

Comment:
Maybe it is better to be a RuntimeException? It is better to not bother method interfaces.